### PR TITLE
Swift 3, UIButton extension and tests

### PR DESCRIPTION
**Pull request by htinlinn**

### CODE REVIEW

File: Marker/Classes/Markdown/MarkdownString.swift [hunk 1]
```diff
@@ -44,11 +44,11 @@ internal func attributedStringFromParsedString(parsedString: String, tags: [Mark
 // Source: http://stackoverflow.com/questions/25138339/nsrange-to-rangestring-index/30404532#30404532
 private extension String {
     
-    func NSRangeFromRange(range: Range<String.Index>) -> NSRange {
-        let from = String.UTF16View.Index(range.startIndex, within: utf16)
-        let to = String.UTF16View.Index(range.endIndex, within: utf16)
+    func NSRangeFromRange(_ range: Range<String.Index>) -> NSRange {
```
Comment (ThibaultKlein) [inline Marker/Classes/Markdown/MarkdownString.swift]: is `NSRangeFromRange` a custom function name or are you overriding something? Otherwise I'll rename `range(from range:)`

File: Marker/Classes/TextStyle.swift [hunk 2]
```diff
@@ -66,7 +66,7 @@ public struct TextStyle {
         
         attributes[NSFontAttributeName] = font
         attributes[NSForegroundColorAttributeName] = textColor
-        attributes[NSKernAttributeName] = characterSpacing
+        attributes[NSKernAttributeName] = characterSpacing as NSObject?
```
Comment (ThibaultKlein) [inline Marker/Classes/TextStyle.swift:69]: why is that required? Also shouldn't it be `as? NSObject`?

Comment (htinlinn) [reply Marker/Classes/TextStyle.swift:69]: I must've accepted whatever the Xcode migrator gave me. Updated now.

File: Marker/Classes/Markdown/MarkdownString.swift (hunk 1 above)
Comment (htinlinn) [reply Marker/Classes/Markdown/MarkdownString.swift]: Renamed it.

**Review by ThibaultKlein — APPROVED** [review]

File: Marker/Classes/TextStyle.swift (hunk 2 above)
Comment (htinlinn) [reply Marker/Classes/TextStyle.swift:69]: Changing this back to the way it was. `as? NSObject` failed `pod lib lint`.
